packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -32,23 +34,30 @@ if [ -z "$resourceGroupName" ]; then
     exitWithUsageInfo
 fi
 
+appRegistrationName="allyappregistration"
+
+# create the app registration if it doesn't exist
+if ! az ad app show --id $clientId 1>/dev/null; then
+    clientId=$(az ad app create --display-name "$appRegistrationName" --query "appId" -o tsv)
+    echo "id is $clientId"
+    echo "created"
+fi
+
 # Start deployment
 echo "Deploying Function App using ARM template"
 resources=$(az group deployment create \
     --resource-group "$resourceGroupName" \
     --template-file "$templateFilePath" \
+    --parameters '{ "clientId": {"value":"'$clientId'"}}' \
     --query "properties.outputResources[].id" \
     -o tsv)
 
 . "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
 functionAppName=$resourceName
 echo "Successfully deployed Function App '$functionAppName'"
 
-# Add system-assigned managed identity to function app
-
-echo "Adding managed identity to Function App '$functionAppName'"
-principalId=$(az webapp identity assign --name "$functionAppName" --resource-group "$resourceGroupName" --query principalId -o tsv)
-echo "Successfully Added managed identity to Function App '$functionAppName'"
+# add the reply url to the app registration reply urls
+az ad app update --id $clientId   --add replyUrls "https://${functionAppName}.azurewebsites.net/.auth/login/aad/callback"
```

Comment:
sudo mv microsoft.gpg /etc/apt/trusted.gpg.d/microsoft.gpg
this does not work on Mac

Reply:
out of this PR scope but, it works for Ubuntu which is our target environment, it doesn't work for Windows or Mac, but if you're using some Ubunto distribution on your system like Ubuntu on Windows, it should work
look at this https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local#v2